src/Geom_Poly.f90

```diff
@@ -361,20 +373,141 @@ ELEMENTAL FUNCTION isCircle_PolygonType(thisPoly) RESULT(bool)
     ENDFUNCTION isCircle_PolygonType
 !
 !-------------------------------------------------------------------------------
+!> @brief Determines whether a polygon meets the criteria for being an arc
+!>        sector. In short, the polygon must have exactly two quadratic edges
+!>        and exactly two non-quadratic edges. A circle is not an arc sector.
+!> @param thisPoly The polygon type to query
+!> @param bool The logical result of the operation
+!>
+    ELEMENTAL FUNCTION isSector_PolygonType(thisPoly) RESULT(bool)
+      CLASS(PolygonType),INTENT(IN) :: thisPoly
+      LOGICAL(SBK) :: bool
+      INTEGER(SIK) :: nNonQuadEdges!,i
+      ! REAL(SRK) :: dist,dist_old
+      bool=.FALSE.
+      IF(thisPoly%isinit) THEN
+        !nVert should be equal to the total number of edges
+        nNonQuadEdges=thisPoly%nVert-thisPoly%nQuadEdge
+        bool=(thisPoly%nQuadEdge == 2) .AND. (nNonQuadEdges == 2)
+        !The following code can be used to check if the lengths of the
+        !non-quadratic edges are equal. This might become relevant if the
+        !sector is required to be "ideal". Leaving commented now for speed.
+        ! dist_old=-1.0_SRK
+        ! IF(ALLOCATED(thisPoly%quad2edge)) THEN
+        !   DO i=1,thisPoly%nVert
+        !     IF(.NOT.ANY(thisPoly%quad2edge == i)) THEN
+        !       dist=Distance(thisPoly%vert(thisPoly%edge(1,i)), &
+        !         thisPoly%vert(thisPoly%edge(2,i)))
+        !       IF(dist_old > 0.0_SRK) THEN
+        !         bool=bool .AND. (dist .APPROXEQA. dist_old)
+        !       ELSE
+        !         dist_old=dist
+        !       ENDIF
+        !     ENDIF
+        !   ENDDO
+        ! ENDIF
+      ENDIF
+    ENDFUNCTION isSector_PolygonType
+!
+!-------------------------------------------------------------------------------
+!> @brief Determines whether a polygon meets the criteria for being an circle
+!>        section. In short, the polygon must have exactly one quadratic edge
+!>        and exactly two non-quadratic edges, or be a full circle.
+!> @param thisPoly The polygon type to query
+!> @param bool The logical result of the operation
+!>
+    ELEMENTAL FUNCTION isSection_PolygonType(thisPoly) RESULT(bool)
+      CLASS(PolygonType),INTENT(IN) :: thisPoly
+      LOGICAL(SBK) :: bool
+      INTEGER(SIK) :: nNonQuadEdges!,i
+      ! REAL(SRK) :: dist,dist_old
+      bool=.FALSE.
+      IF(thisPoly%isinit) THEN
```

Comment:
DBC?

Reply:
Yes, I will try `REQUIRE`ing that the poly is initialized and hopefully it doesn't break other things. Other checking routines like `%isCircle` did not have it so I didn't originally have DBC.